File: cpp_solutions/2275. Largest Combination With Bitwise AND Greater Than Zero.cpp
class Solution {
public:
    int largestCombination(vector<int>& candidates) {
        int ans = 0;
        
        //cout << (3 & 5);
        
        for(int x = 1; x<=10000000; x <<= 1){
            int temp = 0;
            for(int y = 0; y < candidates.size();y++){
                
                if (x & candidates[y]){
                    temp++;
                }
            }
            ans = max(ans, temp);
        }
        
        return ans;
        
    }
};

File: python3_solution_set3/2275. Largest Combination With Bitwise AND Greater Than Zero.py
class Solution:
    def largestCombination(self, candidates: List[int]) -> int:

        can = [format(x,'b') for x in candidates]
        maxs = max([len(x) for x in can])

        for x in range(len(can)):
            diff = maxs - len(can[x])
            can[x] = "".join(["0" for i in range(diff)]) + can[x]

        ans = {x:0 for x in range(maxs)}

        for x in range(len(can)):
            for y in range(len(can[x])):
                if can[x][y] == '1':
                    ans[y]+=1

        
        return max(ans.values())

        
